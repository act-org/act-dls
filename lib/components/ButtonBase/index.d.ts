/**
 * @prettier
 */
import { ButtonProps } from '@material-ui/core';
export declare type Props = ButtonProps;
declare const ButtonBase: import("@material-ui/core").ExtendButtonBase<import("@material-ui/core").ButtonTypeMap<{}, "button">>;
export default ButtonBase;
//# sourceMappingURL=index.d.ts.map