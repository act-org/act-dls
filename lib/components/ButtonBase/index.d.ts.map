{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/ButtonBase/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,EAAU,WAAW,EAAE,MAAM,mBAAmB,CAAC;AAExD,oBAAY,KAAK,GAAG,WAAW,CAAC;AAEhC,QAAA,MAAM,UAAU,uGAAS,CAAC;AAE1B,eAAe,UAAU,CAAC"}