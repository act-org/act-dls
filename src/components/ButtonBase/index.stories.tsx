/**
 * @prettier
 */

import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';

import ButtonBase from '.';

export const base = (): React.ReactElement<any> => (
  <ButtonBase onClick={action('onClick()')}>
    {text('Text', 'Click Me')}
  </ButtonBase>
);

export default {
  component: ButtonBase,
  parameters: {
    info: {
      text: 'This is the base variant of the Button component.',
    },
  },
  title: 'Atoms/Button',
};
