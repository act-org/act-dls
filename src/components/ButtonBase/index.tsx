/**
 * @prettier
 */

import * as React from 'react';
import { Button, ButtonProps } from '@material-ui/core';

export type Props = ButtonProps;

const ButtonBase: React.FC<Props> = ({
  children,
  ...props
}: Props): React.ReactElement<any> => (
  <Button {...props}>{children || 'Button'}</Button>
);

export default ButtonBase;
