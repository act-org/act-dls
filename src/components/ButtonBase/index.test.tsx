/**
 * @prettier
 */

import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { noop } from 'lodash';

import ButtonBase from '.';

describe('ButtonBase', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(<ButtonBase onClick={noop}>Button Label</ButtonBase>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
