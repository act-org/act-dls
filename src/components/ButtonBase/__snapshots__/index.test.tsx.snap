// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBase matches the snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Button Label
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
